src/main/java/com/zendesk/maxwell/schema/TableColumnList.java

```diff
@@ -2,36 +2,40 @@
 
 import java.util.*;
 
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Interner;
+import com.google.common.collect.Interners;
 import com.zendesk.maxwell.schema.columndef.ColumnDef;
 
 
 public class TableColumnList implements Iterable<ColumnDef> {
-	private final List<ColumnDef> columns;
-	private Set<String> columnNames;
+	// reduce count of duplicate ArrayLists/Sets for column lists by providing mutability for the class
+	// through references to an internal immutable object that gets interned. This greatly reduces overhead for
+	// table definitions that are duplicated across databases
+	private InternalImmutableTableColumnList internalState;
```

Comment:
I'd like a better name here than `internalState`.  just `columnList` will be fine.